### PR TITLE
chore(test): test-tabs-native-container-style test screen with scenario

### PR DESCRIPTION
**Pull request by LKuchno**
## Description

Adds a new test screen with manual scenario (`test-tabs-native-container-style`) that exercises the `nativeContainerStyle` prop on `TabsHost`. The scenario allows the user to pick a container background color (unset / blue / yellow / purple) via a `SettingsPicker` and observe the native container styling behavior on both iOS and Android.

Platform surfaces covered: on iOS the color is applied to the `UITabBarController`'s root view (behind both the content area and the tab bar); on Android it is applied to the `FrameLayout` wrapping the focused screen and the `BottomNavigationView`. The `scenario.md` documents platform-specific rendering differences between iOS 18, iOS 26, and Android API 36, as well as tab-switching persistence and rapid cycling edge-case steps.

Closes: https://github.com/software-mansion/react-native-screens-labs/issues/1412

## Changes

- **Test/Tabs**: Added `test-tabs-native-container-style` scenario directory with a two-tab layout (Config + Transparent) covering background color selection via `SettingsPicker` for both iOS and Android. First tab has tabBarBackgroundColor set to Red (Android and iOS 18 and lower), Transparent tab has tabBarBackgroundColor set to transparent (Android) or not set (iOS 18 and lower -> no scrollEdgeAppearance falls into transparent)
- **Test/Tabs**: Added `scenario-description.ts` registering the scenario for both platforms with `e2eCoverage: 'incomplete'` and `smokeTest: false`
- **Test/Tabs**: Added `scenario.md` with full manual test steps.
- **Test/Tabs**: Registered `TestTabsNativeContainerStyle` in `apps/src/tests/single-feature-tests/tabs/index.ts`

### Test visual documentation

|Android lvl 36 | iOS 18.6 | iOS 26.5|
|---|---|---|
| <video src="https://github.com/user-attachments/assets/49979806-b05b-4f78-b799-f3e8192ca2f7"/>| <video src="https://github.com/user-attachments/assets/df9f9330-eacd-4095-a682-bfa029cd9ac6"/>|<video src="https://github.com/user-attachments/assets/6d47fb01-5039-4866-8138-28018eb0bf30"/>

### CODE REVIEW

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+manually.
+
+## Prerequisites
+
+- iOS device or simulator
+- Android emulator or device
+
+## Note
+
+- The color change applies immediately upon selecting a new value; no
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: Is this note necessary? Isn't the immediate change always expected and we should only note this if the change requires restart?

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+### Setting backgroundColor to blue and yellow
+
+2. On the **Config** tab, set the `backgroundColor` picker to `blue`.
+
+- [ ] The picker displays `blue`. The native container's background immediately changes to blue.
+- [ ] **iOS 18:** The blue color is visible behind both the tab content area and the tab bar area, as tabBarBackgroundColor is transparent.
+- [ ] **iOS 26:** The blue color is visible behind the tab content area and through the liquid glass tab bar.
+- [ ] **Android:** The blue color is visible in the area surrounding the focused content screen. The tab bar retains the system default background color.
+
+3. Tap the **Tab** tab in the tab bar.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: nit: we can rename the second tab because "Tap the Tab tab" sounds funny, but we can keep it as well - up to you.

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+## Note
+
+- The color change applies immediately upon selecting a new value; no
+  restart is required.
+- On iOS the color fills the area behind the content screen and around
+  the tab bar (the full `UITabBarController` view). The tab bar itself
+  may partially obscure the color unless the tab bar has a transparent
+  or semi-transparent background.
+- On Android the color fills the `FrameLayout` container, which is
+  visible in the gutters around the focused screen content and behind
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: > in the gutters around the focused screen

I'm not sure what you mean here, the entire background changes color apart from non-translucent tab bar, right? So what are the "gutters" you mention here?

Image

**Review by kligarski — APPROVED** [review]

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md (hunk 1 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: I removed it and refactored note section with information about tab bar bg color behavior across the platform - this made me to refactor screen to check behavior with transparent and no transparent background settings for android and ios18 and lower
https://github.com/software-mansion/react-native-screens/pull/4134/changes/d23fe9078e2262c6f5737a34130b9458616b33d4

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md (hunk 2 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: Renamed to 'Transparent' as after screen refactor it make sense
https://github.com/software-mansion/react-native-screens/pull/4134/changes/d23fe9078e2262c6f5737a34130b9458616b33d4
https://github.com/software-mansion/react-native-screens/pull/4134/changes/d47ab69b93e4666fe9e0ea33f785587d5f02c128

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      ...DEFAULT_TAB_ROUTE_OPTIONS,
+      title: 'Transparent',
+      android: {
+        ...DEFAULT_TAB_ROUTE_OPTIONS.android,
+        standardAppearance: {
+          tabBarBackgroundColor: 'transparent',
+          tabBarItemActiveIndicatorEnabled: false,
+        },
+      },
+    },
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx:140]: There is no need as if scrollEdge  is `undefined`, UIKit uses `standardAppearance`, modified to have a transparent background.

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    Component: ConfigScreen,
+    options: {
+      ...DEFAULT_TAB_ROUTE_OPTIONS,
+      title: 'Config',
+      ios: {
+        ...DEFAULT_TAB_ROUTE_OPTIONS.ios,
+        scrollEdgeAppearance: {
+          tabBarBackgroundColor: Colors.RedDark100,
+        },
+      },
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx:117]: https://github.com/software-mansion/react-native-screens/pull/4134/changes/59a2fd814550e40f3b0cad56639811ec3738c519

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    marginBottom: 4,
+  },
+  value: {
+    fontSize: 13,
+    color: Colors.OffNavy,
+    fontFamily: 'Menlo',
+    backgroundColor: Colors.OffWhite,
+    paddingHorizontal: 8,
+    paddingVertical: 4,
+  },
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/index.tsx]: https://github.com/software-mansion/react-native-screens/pull/4134/changes/b9cc395f55ca4a676d722ebc3ebfd91a3f4a5634

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+5. Set the `backgroundColor` picker to `yellow`.
+
+- [ ] The yellow color is visible behind the tab content area.
+- [ ] The tab bar retains red color.
+- [ ] **iOS 26:** The yellow color is visible through the liquid glass tab bar.
+
+6. Tap the **Transparent** tab, observe the background, then return to **Config**.
+
+- [ ] The container background remains yellow.
+- [ ] The tab bar is transparent, making the blue background visible behind the tabs.
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: https://github.com/software-mansion/react-native-screens/pull/4134/changes/48deb26261f9f47f1d05cc38b822e473607d85a8

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 8]
```diff
@@ -25,15 +25,14 @@ manually.
...
-  may partially obscure the color unless the tab bar has a transparent
-  or semi-transparent background.
-- On Android the color fills the `FrameLayout` container, which is
-  visible in the gutters around the focused screen content and behind
-  the `BottomNavigationView`.
+  the tab bar (the full `UITabBarController` view).
+- On Android the color fills the `FrameLayout` that wraps currently focused
+screen and the `BottomNavigationView`.
+- On Android and iOS 18 and earlier, the tab bar may obscures the color unless
+  a transparent or semi-transparent background is configured.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]:
```suggestion
- On Android and iOS 18 and earlier, the tab bar may obscure the native container
  background color unless a transparent or semi-transparent tab bar background
  color is configured.
```

File: apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md [hunk 9]
```diff
@@ -42,57 +41,60 @@ manually.
...
 
 2. On the **Config** tab, set the `backgroundColor` picker to `blue`.
 
-- [ ] The picker displays `blue`. The native container's background immediately changes to blue.
-- [ ] **iOS 18:** The blue color is visible behind both the tab content area and the tab bar area, as tabBarBackgroundColor is transparent.
-- [ ] **iOS 26:** The blue color is visible behind the tab content area and through the liquid glass tab bar.
-- [ ] **Android:** The blue color is visible in the area surrounding the focused content screen. The tab bar retains the system default background color.
+- [ ] The picker displays `blue`. The native container's background changes to blue.
+- [ ] The blue color is visible behind the tab content area.
+- [ ] The tab bar retains red color.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]:
```suggestion
- [ ] **Android/iOS 18:** The tab bar retains red color.
```

Comment (kligarski) [reply apps/src/tests/single-feature-tests/tabs/test-tabs-native-container-style/scenario.md]: Applies to others as well.

**Review by kligarski — APPROVED** [review]
